climada/util/api_client.py

```diff
@@ -595,20 +574,17 @@ def get_hazard(self, hazard_type, dump_dir=SYSTEM_DIR, max_datasets=1, **kwargs)
         if not hazard_type in HAZ_TYPES:
             raise ValueError("Valid hazard types are a subset of CLIMADA hazard types."
                              f" Currently these types are supported: {HAZ_TYPES}")
-        datasets = self.get_datasets(data_type=hazard_type, **kwargs)
-
-        self._check_datasets_for_concatenation(datasets, max_datasets)
+        dataset = self.get_dataset_info(data_type=hazard_type, **kwargs)
+        return self.to_hazard(dataset, dump_dir)
 
-        return self.to_hazard(datasets, dump_dir)
-
-    def to_hazard(self, datasets, dump_dir=SYSTEM_DIR):
+    def to_hazard(self, dataset, dump_dir=SYSTEM_DIR):
```

Comment:
Should this maybe be a private method? Or what is the goal of this method when compare to `get_hazard`?

Reply:
I wondered too. The use case is: you have already all dataset info objects collected, the files are downloaded and you're ready to do calcuate the impact, but then someone turns off the WLAN. 😏\
After all them method is a short cut that skips a GET request.